src/components/Contacts/ContactDetails/ContactDetailsTab/People/Items/PersonModal/PersonModal.tsx

```diff
@@ -168,70 +180,150 @@ export const EditPersonModal: React.FC<EditPersonModalProps> = ({
     };
   });
 
-  const personFacebookAccounts = person.facebookAccounts.nodes.map(
+  const personFacebookAccounts = person?.facebookAccounts.nodes.map(
     (account) => ({
       id: account.id,
       username: account.username,
       destroy: false,
     }),
   );
-  const personTwitterAccounts = person.twitterAccounts.nodes.map((account) => ({
-    id: account.id,
-    screenName: account.screenName,
-    destroy: false,
-  }));
-  const personLinkedinAccounts = person.linkedinAccounts.nodes.map(
+
+  const personTwitterAccounts = person?.twitterAccounts.nodes.map(
+    (account) => ({
+      id: account.id,
+      screenName: account.screenName,
+      destroy: false,
+    }),
+  );
+
+  const personLinkedinAccounts = person?.linkedinAccounts.nodes.map(
     (account) => ({
       id: account.id,
       publicUrl: account.publicUrl,
       destroy: false,
     }),
   );
-  const personWebsites = person.websites.nodes.map((account) => ({
+
+  const personWebsites = person?.websites.nodes.map((account) => ({
     id: account.id,
     url: account.url,
     destroy: false,
   }));
 
-  const initialPerson: PersonUpdateInput = {
-    id: person.id,
-    firstName: person.firstName,
-    lastName: person.lastName,
-    title: person.title,
-    suffix: person.suffix,
-    phoneNumbers: personPhoneNumbers,
-    emailAddresses: personEmails,
-    optoutEnewsletter: person.optoutEnewsletter,
-    birthdayDay: person.birthdayDay,
-    birthdayMonth: person.birthdayMonth,
-    birthdayYear: person.birthdayYear,
-    maritalStatus: person.maritalStatus,
-    gender: person.gender,
-    anniversaryDay: person.anniversaryDay,
-    anniversaryMonth: person.anniversaryMonth,
-    anniversaryYear: person.anniversaryYear,
-    almaMater: person.almaMater,
-    employer: person.employer,
-    occupation: person.occupation,
-    facebookAccounts: personFacebookAccounts,
-    twitterAccounts: personTwitterAccounts,
-    linkedinAccounts: personLinkedinAccounts,
-    websites: personWebsites,
-    legalFirstName: person.legalFirstName,
-    deceased: person.deceased,
-  };
+  const initialPerson: PersonCreateInput | PersonUpdateInput = person
+    ? {
+        id: person.id,
```

Comment:
Couldn't something like this have {...person} to get all these properties? A lot of them look similar to PersonUpdateInput

Reply:
Ahh yep good idea I could definitely do ```...person,``` and then just keep the fields that I need to specifically set like ```phoneNumbers```, ```emailAddresses```, etc.

Reply:
Strange but for some reason, whenever I used the spread it seems to work fine but all the tests fail 🤷‍♂️ I may just leave as-is for now.